docs/DeveloperGuide.md

```diff
@@ -1,34 +1,630 @@
+respective method.
+> ✔️ `Expense` type invokes `Ui#printTotalAmountExpense()`
+
+> ✔️ `Loan` type invokes `Ui#printTotalAmountLoan()`
+
+> ✔️ `Saving` type invokes `Ui#printTotalAmountSaving()`
+
+***Step 4:***\
+The `Ui` will handle the respective invocation call. The basis for the three methods utilizes the `for` loop to
+iterate through the `recordList` and will only add to the `totalAmount` if it is an instance of the respective
+record type.
+> 📝 The `Ui#printTotalAmountLoan()` will imposed additional check on whether the record is returned.
+> Only records that are not returned will be added to the `totalAmount`
+
+***Step 5:***\
+Finally, the `totalAmount` will be rounded off to two decimal place before printing onto the console.
+
+#### 4.3.2 Design Consideration
+
+This section shows the design considerations taken when implementing the view feature.
+
+Aspect: **How many figures should the amount be allowed up to**
+
+Since it is entirely possible that the amount provided by user to be very large or very small, the two choices to
+consider would be:
+* Integer (Classic approach)
+* BigDecimal (Non-classic approach)
+
+|Approach | Pros | Cons| 
+|---------|------|-----|
+|Integer|No additional class required|Limited to a fix amount of digit available to enter|
+|BigDecimal|Allow more digits than classic approach|Need to create a BigDecimal object|
+
+Having considered two of the approaches, we have decided to adopt the second approach. 
+The deterministic factor was the possibility of very large number such as 1000 billions. Thus, covering more than
+what the integer data type provided can cater to higher flexibility of the application.
+
+Aspect: **When to round off the amount to two decimal place**
+
+As the user can enter decimal into the amount field, the `ViewCommand` has two options of when to round off the value:
+* When assigning value to the amount variable
+* When printing the final computed value
+
+|Approach|Pros|Cons|
+|--------|----|----|
+|Assigning value to the amount variable|Round off early to avoid complication|Additional time per assignment to round off|
+|Printing the final computed value|Only round off once|Rounding off might be couple to other method which needs the amount field|
+
+With the two approaches considered, we have decided to adopt the second approach as it is more efficient to round off
+at the end and to preserve the accuracy for two decimal place. Since no other method is accessing the amount variable,
+is it possible avoid rounding off during each value assignment until new or existing features requires it.
+
+### 4.4 Return Feature
+The `return` feature allows Finux users to mark a loan as returned.
+
+`return -i <index of loan to return> -d <return date>`
+
+Upon marking the loan as returned, *creditscore* for the borrower who loaned the book will be updated and saved into `finux.txt`
+
+Example:
+`return -i 2 -d 2021-03-16`
+
+Output:
+Loan marked as returned: [L][2021-03-16] Loan to Tom [v]
```

Comment:
Should this be in a code block?

Reply:
I think that was some sample placeholder for that section, we will renew it. Thanks!